04_numeronym/hsy3418/main.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+//numeronyms implements a function that returns a slice of numeronyms for its input strings
+func numeronyms(vals ...string) []string {
+
+	numeronymsList := vals
```

Comment:
This is dangerous. It is not making a copy of the input so it is actually changing it. In your case, the input is an anonymous slice so you don't notice, but if `numeronyms` was called like this:
```
vals := []string{"accessibility", "Kubernetes", "abc}
fmt.Fprintln(out, numeronyms(vals...))
```
then your function would be changing the contents of the `vals` slice in the caller.

You should be making a new slice for the output value - `result := make([]string, len(vals))` and putting the numeronyms in that, and then returning it.